bigframes/core/compile/compiler.py

```diff
@@ -76,13 +76,22 @@ def _compile_node(
     @_compile_node.register
     def compile_join(self, node: nodes.JoinNode, ordered: bool = True):
         if ordered:
-            left_ordered = self.compile_ordered_ir(node.left_child)
-            right_ordered = self.compile_ordered_ir(node.right_child)
-            return bigframes.core.compile.single_column.join_by_column_ordered(
-                left=left_ordered,
-                right=right_ordered,
-                join=node.join,
-            )
+            if self.strict:
+                left_ordered = self.compile_ordered_ir(node.left_child)
+                right_ordered = self.compile_ordered_ir(node.right_child)
+                return bigframes.core.compile.single_column.join_by_column_ordered(
+                    left=left_ordered,
+                    right=right_ordered,
+                    join=node.join,
+                )
+            else:
+                left_unordered = self.compile_unordered_ir(node.left_child)
```

Comment:
I'm curious if wee need to do anything to handle `sort=True` behavior from the [pandas join](https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.join.html#pandas-dataframe-join)? I'm guessing "no", as I suspect this would be implemented as a sort after the join.

Also, let's add a comment explaining this optimization:

```suggestion
                # In general, joins are an ordering destroying operation.
                # With ordering_mode = "partial", make this explicit. In
                # this case, we don't need to provide a deterministic ordering.
                left_unordered = self.compile_unordered_ir(node.left_child)
```

Reply:
added comment

Reply:
And yes, `sort=True` sorts as an additional operation and so will still apply: https://github.com/googleapis/python-bigquery-dataframes/blob/main/bigframes/core/blocks.py#L2073-L2080